Add nobel prize demo (#4)
meilisearch/meilisearch-vue: examples/nobel-prizes/README.md

[![Nobel prices demo](misc/marie-heavy.gif)](https://nobel.meilisearch.com)

## Run and develop locally

First, you will need [yarn](https://classic.yarnpkg.com/en/docs/install/) or [npm](https://www.npmjs.com/get-npm).</br>
Then run:

```bash
$ git clone git@github.com:meilisearch/meilisearch-vue.git
$ cd meilisearch-vue/examples/nobel-prizes
$ yarn # or npm install
$ yarn serve # or npm run serve
$ open http://localhost:8080/
```

We've included some credentials in the code, allowing you to test the demo without having to create your own MeiliSearch instance.

## Data import

If you want to replicate this demo using your own MeiliSearch, you will have to follow these steps:

### 1. Download and Launch

[Download and launch MeiliSearch](https://docs.meilisearch.com/guides/advanced_guides/installation.html) using the basic configuration. For example, by using [cURL](https://curl.haxx.se/) in your terminal.

```bash
$ curl -L https://install.meilisearch.com | sh
$ ./meilisearch
Server is listening on: http://127.0.0.1:7700
```

> This walkthrough does not uses the [master key](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#master-key) option.

Reviewer: ```suggestion
This walkthrough does not use the [master key](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#master-key) option.
```

### 2. Run the setup

Go inside the `setup` directory and run the following:
```bash
$ cd setup/
$ yarn # or npm install
$ yarn start # or npm run start
```

This will do the following:
- Create an index called `prizes` in your MeiliSearch instance.
- Add all Nobel prizes documents to that index.
- Add custom settings for a more relevant search.

### 3. Change the credentials in the front

The credentials of the MeiliSearch instance are written inside `src/meilisearch-client.js`. By default, these are the public credentials for the Nobel dataset.

You have to change these credentials with your local instance:

```javascript
const MEILISEARCH_HOST = "http://127.0.0.1:7700"
const MEILISEARCH_API_KEY = ""

> // Because we did not set any API key at [step 1](#1.-download-and-launch), we don't need to provide any `MEILISEARCH_API_KEY` here.

Reviewer: Putting this sentence as a comment makes the link not working (this link is not mandatory from my POV).
Plus it looks like this comment is linked to the code below, but it concerns more the `const MEILISEARCH_API_KEY = ""` above.

Author: I moved the whole comment outside the code block on the bottom.

## MeiliSearch Search in Nobel Winners

In this directory, you will find the front end to search through all Nobel winners and their prizes' specifics.
The search is powered by [MeiliSearch](https://github.com/meilisearch/meilisearch), a powerful, fast, open-source, easy to use and deploy search engine.

> ## Demo 

Reviewer: ```suggestion
## Demo

```

<!-- TODO: CHANGE LINK -->
_Based on [How to Search Nobel Prize Winners Faster With MeiliSearch and JavaScript](https://blog.meilisearch.com/p/8b7238e0-eb2f-47ee-90d4-815a8357f12b/)._

[![Nobel prices demo](misc/marie-heavy.gif)](https://nobel.meilisearch.com)

> 

Reviewer: ```suggestion
```

## Run and develop locally

> First, you will need [yarn](https://classic.yarnpkg.com/en/docs/install/) or [npm](https://www.npmjs.com/get-npm).

Reviewer: ```suggestion
First, you will need [yarn](https://classic.yarnpkg.com/en/docs/install/) or [npm](https://www.npmjs.com/get-npm).</br>
```

Then run:

```bash

> git clone git@github.com:meilisearch/meilisearch-vue.git
> cd meilisearch-vue/examples/nobel-prizes
> yarn # or npm install
> yarn serve # or npm run serve
> open http://localhost:8080/

Reviewer: ```suggestion
$ git clone git@github.com:meilisearch/meilisearch-vue.git
$ cd meilisearch-vue/examples/nobel-prizes
$ yarn # or npm install
$ yarn serve # or npm run serve
$ open http://localhost:8080/
```

```

We've included some credentials in the code, allowing you to test the demo without having to create your own MeiliSearch instance.

> 
> ## Data import

Reviewer: ```suggestion
## Data import

```

If you want to replicate this demo using your own MeiliSearch, you will have to follow these steps: 

### 1. Download and Launch 

[Download and launch MeiliSearch](https://docs.meilisearch.com/guides/advanced_guides/installation.html) using the basic configuration.

> Do not add any master key option. This setup **works without api key**. If you would like to add an `api key`, you can add the api key on line 8 of `setup.js`:

Reviewer: A little bit confusing. We are talking about "master key option" (maybe talking about `--master-key` flag with a redirection to the docs), and then about API key.
We ask not to add the API key, and then, we say we could want to add one.
I personally understand what it's written here, but because I know MeiliSearch 🙂

Reviewer: Maybe just the simple `cURL` example would be simple and a redirection to the docs to set credentials would be enough?
This way we "enforce" the first reader to try without API key first and to be able to follow the step 3.

```javascript
// setup/setup.js
const client = new MeiliSearch({
    host: 'http://127.0.0.1:7700'
    apiKey: "my_api_key"
})
```

### 2. Run the setup

Go inside the `setup` directory and run the following:
```bash

> cd setup/
> yarn # or npm install
> yarn start # or npm run start

Reviewer: ```suggestion
$ cd setup/
$ yarn # or npm install
$ yarn start # or npm run start
```

```

This will do the following: 

> - Create an index called `prizes` in your MeiliSearch instance

Reviewer: ```suggestion
- Create an index called `prizes` in your MeiliSearch instance.
```

- Add all Nobel prizes documents to that index.

> - Add custom settings for a more relevant search

Reviewer: ```suggestion
- Add custom settings for a more relevant search.
```

### 3. Change the credentials in the front

The credentials of the MeiliSearch instance are written inside `src/meilisearch-client.js`. By default, these are the public credentials for the Nobel dataset. 

You have to change these credentials with your local instance:

```javascript
const MEILISEARCH_HOST = "http://127.0.0.1:7700"
const MEILISEARCH_API_KEY = ""

> 

Reviewer: ```suggestion
Because we did not set any API key at [step 1](#1.-download-and-launch), we don't need to provide any `MEILISEARCH_API_KEY` here.
```

export {
  MEILISEARCH_HOST,
  MEILISEARCH_API_KEY
}
```

### 4. Run the project

> You can now go back to the root directory and run the project. The front-end is now communicating with your MeiliSearch instance.

Reviewer: I like this kind of comforting sentence 👍

```bash

> cd ..
> yarn serve # or npm run serve
> open http://localhost:8080/

Reviewer: ```suggestion
$ cd ..
$ yarn serve # or npm run serve
$ open http://localhost:8080/
```

## MeiliSearch Search in Nobel Winners

> In this directory, you will find the front end to search through all Nobel winners and their prizes' specifics.

Reviewer 2: ```suggestion
In this directory, you will find the front-end interface to search through all Nobel Prize winners and their details.
```

> The search is powered by [MeiliSearch](https://github.com/meilisearch/meilisearch), a powerful, fast, open-source, easy to use and deploy search engine.

Reviewer 2: ```suggestion
The search is powered by [MeiliSearch](https://github.com/meilisearch/meilisearch), a powerful, fast and open-source search engine easy to use and deploy.
```

Author: If you want you can make the same change in the readme of meilisearch! https://github.com/meilisearch/meilisearch

## Demo

<!-- TODO: CHANGE LINK -->
_Based on [How to Search Nobel Prize Winners Faster With MeiliSearch and JavaScript](https://blog.meilisearch.com/p/8b7238e0-eb2f-47ee-90d4-815a8357f12b/)._

[![Nobel prices demo](misc/marie-heavy.gif)](https://nobel.meilisearch.com)

## Run and develop locally

> First, you will need [yarn](https://classic.yarnpkg.com/en/docs/install/) or [npm](https://www.npmjs.com/get-npm).</br>

Reviewer 2: ```suggestion
First, you will need to install [yarn](https://classic.yarnpkg.com/en/docs/install/) or [npm](https://www.npmjs.com/get-npm).</br>
```

Then run:

```bash
$ git clone git@github.com:meilisearch/meilisearch-vue.git
$ cd meilisearch-vue/examples/nobel-prizes
$ yarn # or npm install
$ yarn serve # or npm run serve

> $ open http://localhost:8080/

Reviewer 2: `open` is a OSX specific command, this isn't meant for OSX developers only so I wouldn't use it

```

We've included some credentials in the code, allowing you to test the demo without having to create your own MeiliSearch instance.

## Data import

> If you want to replicate this demo using your own MeiliSearch, you will have to follow these steps:

Reviewer 2: ```suggestion
If you want to replicate this demo using your own MeiliSearch instance, you will have to follow these steps:
```

> ### 1. Download and Launch

Reviewer 2: ```suggestion
### 1. Download and launch MeiliSearch
```

[Download and launch MeiliSearch](https://docs.meilisearch.com/guides/advanced_guides/installation.html) using the basic configuration. For example, by using [cURL](https://curl.haxx.se/) in your terminal.

```bash
$ curl -L https://install.meilisearch.com | sh
$ ./meilisearch

> Server is listening on: http://127.0.0.1:7700

Reviewer 2: ```suggestion
```

```

> This walkthrough does not use the [master key](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#master-key) option.

Reviewer 2: ```suggestion
This walkthrough runs MeiliSearch in a development environment, and therefore it does not use any [master key](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#master-key).
```

### 2. Run the setup

> Go inside the `setup` directory and run the following:

Reviewer 2: ```suggestion
Go inside the `setup` directory and run the following commands:
```

```bash
$ cd setup/
$ yarn # or npm install
$ yarn start # or npm run start
```

This will do the following:
- Create an index called `prizes` in your MeiliSearch instance.
- Add all Nobel prizes documents to that index.
- Add custom settings for a more relevant search.

> ### 3. Change the credentials in the front

Reviewer 2: ```suggestion
### 3. Change the credentials in the front-end client
```

The credentials of the MeiliSearch instance are written inside `src/meilisearch-client.js`. By default, these are the public credentials for the Nobel dataset.

You have to change these credentials with your local instance:

```javascript
const MEILISEARCH_HOST = "http://127.0.0.1:7700"
const MEILISEARCH_API_KEY = ""

export {
  MEILISEARCH_HOST,
  MEILISEARCH_API_KEY
}
```

Because we did not set any API key at [step 1](#1.-download-and-launch), we don't need to provide any `MEILISEARCH_API_KEY` here.

### 4. Run the project

> You can now go back to the root directory and run the project. The front-end is now communicating with your MeiliSearch instance.

Reviewer 2: ```suggestion
You can now go back to the root directory and run the project. The front-end client is now communicating with your MeiliSearch instance.
```

```bash
$ cd ..
$ yarn serve # or npm run serve
$ open http://localhost:8080/
```

## Working example

> A working example runs on CodeSandBox:

Reviewer 2: ```suggestion
A working example can be tested on CodeSandBox:
```

```bash
$ curl -L https://install.meilisearch.com | sh
$ ./meilisearch
```

This walkthrough runs MeiliSearch in a development environment, and therefore it does not use any [master key](https://docs.meilisearch.com/guides/advanced_guides/configuration.html#master-key).

### 2. Run the setup

Go inside the `setup` directory and run the following commands:
```bash
$ cd setup/
$ yarn # or npm install
$ yarn start # or npm run start
```

This will do the following:
- Create an index called `prizes` in your MeiliSearch instance.
- Add all Nobel prizes documents to that index.
- Add custom settings for a more relevant search.

### 3. Change the credentials in the front-end client

The credentials of the MeiliSearch instance are written inside `src/meilisearch-client.js`. By default, these are the public credentials for the Nobel dataset.

You have to change these credentials with your local instance:

```javascript
const MEILISEARCH_HOST = "http://127.0.0.1:7700"
const MEILISEARCH_API_KEY = ""

export {
  MEILISEARCH_HOST,
  MEILISEARCH_API_KEY
}
```

> Because we did not set any API key at [step 1](#1.-download-and-launch), we don't need to provide any `MEILISEARCH_API_KEY` here.

Reviewer: ```suggestion
Because we did not set any API key at [step 1](#1-download-and-launch), we don't need to provide any `MEILISEARCH_API_KEY` here.
```
